<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>BEAMR ‚Ä¢ BEAMR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="BEAMR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">BEAMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/BEAMR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/annaSeffernick/BEAMR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>BEAMR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/annaSeffernick/BEAMR/blob/main/vignettes/articles/BEAMR.Rmd" class="external-link"><code>vignettes/articles/BEAMR.Rmd</code></a></small>
      <div class="d-none name"><code>BEAMR.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://annaseffernick.github.io/BEAMR/">BEAMR</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Bootstrap Evaluation of Association Matrices (BEAMR) is a method to
integrate multiple forms of omics data with multiple clinical endpoints.
BEAMR uses simple association models and bootstrapping to provide a
consolidated gene ranking using all of the feature/endpoint
associations.</p>
<p>Here is a brief overview of BEAMR. First, define sets that link the
genomic features from each omic dataset to a ‚Äúset‚Äù. This set could be
any meaningful grouping of features, such as gene, pathway, etc. Next,
for each set, compute the association estimate matrix (AEM), which is a
matrix of regression coefficients from univariate association models of
each endpoint with each feature in that set. Then we consider the AEM as
a point in multivariate association estiamte space and determine its
distance from the null point of no association (all entries in the
AEM=0). Finally, we use bootstrap resampling to determine if the
observed set is significantly different from the null. This is
illustrated in Figure 1.</p>
<p><img src="BeamFigure1_PaperLayout_2023-04-20.png" width="360" style="display: block; margin: auto;"></p>
<p>In this vignette, we describe how to use the BEAMR package with a
pediatric T-ALL example dataset from COG trial AALL0434 published in <a href="https://www.nature.com/articles/ng.3909" class="external-link">Liu et al., 2017</a>.
This dataset contains 265 patients, three clinical endpoints, and two
omics.</p>
</div>
<div class="section level2">
<h2 id="the-data">The Data<a class="anchor" aria-label="anchor" href="#the-data"></a>
</h2>
<p>The clinical data are saved in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtext mathvariant="monospace">ùöåùöïùöíùöóùöè</mtext><annotation encoding="application/x-tex">\texttt{clinf}</annotation></semantics></math>.
The three clinical endpoints are minimal residual disease measured at
Day 29 (MRD29), event-free survival (EFS), and overall survival
(OS).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clinf</span><span class="op">)</span></span>
<span><span class="co">#&gt;            ID MRD29 RNA.clm Lesion.clm Lesion.id RNA.id   EFS    OS</span></span>
<span><span class="co">#&gt; PARASZ PARASZ 0.000  PARASZ     PARASZ    PARASZ PARASZ 3087+ 3087+</span></span>
<span><span class="co">#&gt; PARAYM PARAYM 0.000  PARAYM     PARAYM    PARAYM PARAYM 3399+ 3399+</span></span>
<span><span class="co">#&gt; PARCVM PARCVM 0.000  PARCVM     PARCVM    PARCVM PARCVM 2424+ 2424+</span></span>
<span><span class="co">#&gt; PAREGZ PAREGZ 0.490  PAREGZ     PAREGZ    PAREGZ PAREGZ 3087+ 3087+</span></span>
<span><span class="co">#&gt; PARFDL PARFDL 0.000  PARFDL     PARFDL    PARFDL PARFDL 3075+ 3075+</span></span>
<span><span class="co">#&gt; PARFIH PARFIH 0.028  PARFIH     PARFIH    PARFIH PARFIH 1099+ 1099+</span></span></code></pre></div>
<p>There are two omics datasets included here.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtext mathvariant="monospace">ùôªùöéùöúùöíùöòùöó</mtext><annotation encoding="application/x-tex">\texttt{Lesion}</annotation></semantics></math>
is an indicator matrix denoting the presence or absence of a lesion.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtext mathvariant="monospace">ùöÅùôΩùô∞</mtext><annotation encoding="application/x-tex">\texttt{RNA}</annotation></semantics></math>
is a matrix of gene expression.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">omicdat</span><span class="op">$</span><span class="va">Lesion</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;                          PARWNW PASXUC PATXNR PASYIS PATBGC</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000148400_mutation      1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000099810_loss          1      1      1      1      0</span></span>
<span><span class="va">omicdat</span><span class="op">$</span><span class="va">RNA</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;                    PARFIH     PARFPJ     PARFXJ     PARKLK     PARLJA</span></span>
<span><span class="co">#&gt; ENSG00000121410 0.4976379 0.29994795 1.13973142 0.32584690 0.35940872</span></span>
<span><span class="co">#&gt; ENSG00000148584 0.0000000 0.00000000 0.00000000 0.00000000 0.00000000</span></span>
<span><span class="co">#&gt; ENSG00000175899 0.0176369 0.11316706 0.02913480 0.00992041 0.02672925</span></span>
<span><span class="co">#&gt; ENSG00000166535 0.0000000 0.02261638 0.01278318 0.01735184 0.00000000</span></span>
<span><span class="co">#&gt; ENSG00000184389 0.1067503 1.46126661 0.32734246 0.56042150 0.49385225</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-beamr-data">Prepare BEAMR Data<a class="anchor" aria-label="anchor" href="#prepare-beamr-data"></a>
</h2>
<p>To use BEAMR, we need the data in a specific format. This can be
accomplished with the <code><a href="../reference/prep_beam_data.html">prep_beam_data()</a></code> function. In
addition to the clinical and omics data, this function also requires
omic annotation and set data input. The omics annotation has id and gene
columns.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">omicann</span><span class="op">$</span><span class="va">Lesion</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         id            gene</span></span>
<span><span class="co">#&gt; 1     ENSG00000264545_loss ENSG00000264545</span></span>
<span><span class="co">#&gt; 2     ENSG00000147889_loss ENSG00000147889</span></span>
<span><span class="co">#&gt; 3     ENSG00000224854_loss ENSG00000224854</span></span>
<span><span class="co">#&gt; 4 ENSG00000148400_mutation ENSG00000148400</span></span>
<span><span class="co">#&gt; 5     ENSG00000099810_loss ENSG00000099810</span></span>
<span><span class="co">#&gt; 6     ENSG00000264801_loss ENSG00000264801</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">omicann</span><span class="op">$</span><span class="va">RNA</span><span class="op">)</span></span>
<span><span class="co">#&gt;                id            gene</span></span>
<span><span class="co">#&gt; 1 ENSG00000121410 ENSG00000121410</span></span>
<span><span class="co">#&gt; 2 ENSG00000148584 ENSG00000148584</span></span>
<span><span class="co">#&gt; 3 ENSG00000175899 ENSG00000175899</span></span>
<span><span class="co">#&gt; 4 ENSG00000166535 ENSG00000166535</span></span>
<span><span class="co">#&gt; 5 ENSG00000184389 ENSG00000184389</span></span>
<span><span class="co">#&gt; 6 ENSG00000128274 ENSG00000128274</span></span></code></pre></div>
<p>The set data input contains the mapping from the omics features to
the sets. It is a data.frame with columns ‚Äúset.id‚Äù, ‚Äúmtx‚Äù, and ‚Äúrow.id‚Äù.
‚Äúset.id‚Äù is the name of the set, which is Ensembl ID in this example.
‚Äúmtx‚Äù defines the omic type (e.g., ‚ÄúLesion‚Äù or ‚ÄúRNA‚Äù). ‚Äúrow.id‚Äù is the
feature name, which must be an element of the rownames of the omics
matrices.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">setdat</span><span class="op">)</span></span>
<span><span class="co">#&gt;            set.id mtx.id          row.id</span></span>
<span><span class="co">#&gt; 1 ENSG00000121410    RNA ENSG00000121410</span></span>
<span><span class="co">#&gt; 2 ENSG00000148584    RNA ENSG00000148584</span></span>
<span><span class="co">#&gt; 3 ENSG00000175899    RNA ENSG00000175899</span></span>
<span><span class="co">#&gt; 4 ENSG00000166535    RNA ENSG00000166535</span></span>
<span><span class="co">#&gt; 5 ENSG00000184389    RNA ENSG00000184389</span></span>
<span><span class="co">#&gt; 6 ENSG00000128274    RNA ENSG00000128274</span></span></code></pre></div>
<p>With these components, we can now call <code><a href="../reference/prep_beam_data.html">prep_beam_data()</a></code>.
We also specify the number of bootstrap replicates with the ‚Äún.boot‚Äù
arguement and can set the seed for reproducibility. The output is a
beam.data object, which is a list with main.data (clinf), mtx.data
(omicdat), mtx.anns (omicann), anns.mtch (summary of omic annotation),
set.data (setdat), set.anns, and boot.index (indices of bootstrap
resamples).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">beam.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep_beam_data.html">prep_beam_data</a></span><span class="op">(</span>main.data<span class="op">=</span><span class="va">clinf</span>, mtx.data<span class="op">=</span><span class="va">omicdat</span>,</span>
<span>                            mtx.anns<span class="op">=</span><span class="va">omicann</span>, set.data<span class="op">=</span><span class="va">setdat</span>,</span>
<span>                            set.anns<span class="op">=</span><span class="cn">NULL</span>, n.boot<span class="op">=</span><span class="fl">10</span>, seed<span class="op">=</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="co">#&gt; Checking inputs: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt;   Checking that each element of mtx.data is a matrix: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt;   Checking that each element of mtx.anns is a data.frame: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt; Aligning main.data with each mtx.data: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt;   Working on mtx.data Lesion (1 of 2): Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt;   Working on mtx.data RNA (2 of 2): Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt; Warning in get_id_index(main.data, mtx.clms): Some ids not matched; returning</span></span>
<span><span class="co">#&gt; NAs for those.</span></span>
<span><span class="co">#&gt;   Working on mtx.anns: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt;   Matching matrix 1 with annotations: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt;   Matching matrix 2 with annotations: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt;   Checking set.data: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt;     Ordering and indexing set.data: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt;     Checking section 1 of 40 of set.data: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt; Generating bootstrap index matrix: Tue Jul  9 18:41:27 2024</span></span>
<span><span class="co">#&gt; Packaging and returning result: Tue Jul  9 18:41:27 2024</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">beam.data</span></span>
<span><span class="co">#&gt; main.data: 265 rows and 8 columns. </span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;            ID MRD29 RNA.clm Lesion.clm Lesion.id</span></span>
<span><span class="co">#&gt; PARASZ PARASZ  0.00      46         21    PARASZ</span></span>
<span><span class="co">#&gt; PARAYM PARAYM  0.00      47        108    PARAYM</span></span>
<span><span class="co">#&gt; PARCVM PARCVM  0.00      15         84    PARCVM</span></span>
<span><span class="co">#&gt; PAREGZ PAREGZ  0.49     170        142    PAREGZ</span></span>
<span><span class="co">#&gt; PARFDL PARFDL  0.00     184        175    PARFDL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; mtx.data: </span></span>
<span><span class="co">#&gt;   mtx.data Lesion: 265 columns linked to 265 rows of main.data. </span></span>
<span><span class="co">#&gt;   mtx.data RNA: 264 columns linked to 264 rows of main.data. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Lesion: </span></span>
<span><span class="co">#&gt;                          PARWNW PASXUC PATXNR PASYIS PATBGC</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000148400_mutation      1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000099810_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; RNA: </span></span>
<span><span class="co">#&gt;                    PARFIH     PARFPJ     PARFXJ     PARKLK     PARLJA</span></span>
<span><span class="co">#&gt; ENSG00000121410 0.4976379 0.29994795 1.13973142 0.32584690 0.35940872</span></span>
<span><span class="co">#&gt; ENSG00000148584 0.0000000 0.00000000 0.00000000 0.00000000 0.00000000</span></span>
<span><span class="co">#&gt; ENSG00000175899 0.0176369 0.11316706 0.02913480 0.00992041 0.02672925</span></span>
<span><span class="co">#&gt; ENSG00000166535 0.0000000 0.02261638 0.01278318 0.01735184 0.00000000</span></span>
<span><span class="co">#&gt; ENSG00000184389 0.1067503 1.46126661 0.32734246 0.56042150 0.49385225</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; mtx.anns: </span></span>
<span><span class="co">#&gt;   Lesion: 20 rows and 2 columns. </span></span>
<span><span class="co">#&gt;   RNA: 20 rows and 2 columns. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Lesion: </span></span>
<span><span class="co">#&gt;                         id            gene</span></span>
<span><span class="co">#&gt; 1     ENSG00000264545_loss ENSG00000264545</span></span>
<span><span class="co">#&gt; 2     ENSG00000147889_loss ENSG00000147889</span></span>
<span><span class="co">#&gt; 3     ENSG00000224854_loss ENSG00000224854</span></span>
<span><span class="co">#&gt; 4 ENSG00000148400_mutation ENSG00000148400</span></span>
<span><span class="co">#&gt; 5     ENSG00000099810_loss ENSG00000099810</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; RNA: </span></span>
<span><span class="co">#&gt;                id            gene</span></span>
<span><span class="co">#&gt; 1 ENSG00000121410 ENSG00000121410</span></span>
<span><span class="co">#&gt; 2 ENSG00000148584 ENSG00000148584</span></span>
<span><span class="co">#&gt; 3 ENSG00000175899 ENSG00000175899</span></span>
<span><span class="co">#&gt; 4 ENSG00000166535 ENSG00000166535</span></span>
<span><span class="co">#&gt; 5 ENSG00000184389 ENSG00000184389</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; anns.mtch: </span></span>
<span><span class="co">#&gt;   mtx.data mtx.anns id.clm nrow.mtx nrow.ann nrow.map</span></span>
<span><span class="co">#&gt; 1   Lesion   Lesion     id       20       20       20</span></span>
<span><span class="co">#&gt; 2      RNA      RNA     id       20       20       20</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; set.data: 40 rows assigning sets to data.mtx rows. </span></span>
<span><span class="co">#&gt;             set.id mtx.id               row.id</span></span>
<span><span class="co">#&gt; 9  ENSG00000081760    RNA      ENSG00000081760</span></span>
<span><span class="co">#&gt; 8  ENSG00000094914    RNA      ENSG00000094914</span></span>
<span><span class="co">#&gt; 25 ENSG00000099810 Lesion ENSG00000099810_loss</span></span>
<span><span class="co">#&gt; 19 ENSG00000099810    RNA      ENSG00000099810</span></span>
<span><span class="co">#&gt; 14 ENSG00000109576    RNA      ENSG00000109576</span></span>
<span><span class="co">#&gt; 10 ENSG00000114771    RNA      ENSG00000114771</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; set.anns:  rows of set annotations.</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; boot.index: 11 rows and 265 columns of bootstrap indices. </span></span>
<span><span class="co">#&gt;      [,1] [,2] [,3] [,4] [,5]</span></span>
<span><span class="co">#&gt; [1,]    1    2    3    4    5</span></span>
<span><span class="co">#&gt; [2,]  179   14  195  118  229</span></span>
<span><span class="co">#&gt; [3,]  108    8  114  261   29</span></span>
<span><span class="co">#&gt; [4,]   55   19  241  218  155</span></span>
<span><span class="co">#&gt; [5,]  145  200  211   69   46</span></span></code></pre></div>
<p>Before calculating the association estimate matrix, we need to
specify the models to fit. We do this through ‚Äúbeam.specs‚Äù which can be
generated automatically using <code><a href="../reference/prep_beam_specs.html">prep_beam_specs()</a></code>. This
function defines the model form based on the endpoint type. The
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtext mathvariant="monospace">ùöèùöíùöõùöùùöë=ùöÉùöÅùöÑùô¥</mtext><annotation encoding="application/x-tex">\texttt{firth=TRUE}</annotation></semantics></math>
option uses Firth penalized Cox models which are useful in the presence
of rare events or omics.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">beam.specs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep_beam_specs.html">prep_beam_specs</a></span><span class="op">(</span>beam.data<span class="op">=</span><span class="va">beam.data</span>, endpts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MRD29"</span>, <span class="st">"OS"</span>, <span class="st">"EFS"</span><span class="op">)</span>, firth<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; MRD29 is continuous, fitting lm</span></span>
<span><span class="co">#&gt; OS is survival endpoint, fitting coxphf2</span></span>
<span><span class="co">#&gt; EFS is survival endpoint, fitting coxphf2</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">beam.specs</span></span>
<span><span class="co">#&gt;           name    mtx                                         mdl</span></span>
<span><span class="co">#&gt; 1 Lesion.MRD29 Lesion    lm(MRD29~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 2    RNA.MRD29    RNA    lm(MRD29~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 3    Lesion.OS Lesion  coxphf2(OS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 4       RNA.OS    RNA  coxphf2(OS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 5   Lesion.EFS Lesion coxphf2(EFS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 6      RNA.EFS    RNA coxphf2(EFS~mtx.row,data=main.data,model=T)</span></span></code></pre></div>
<p>Users may also specify their own models by entering an optional
‚Äúendptmdl‚Äù argument, which is a data.frame with string columns called
‚Äúendpt‚Äù and ‚Äúmdl‚Äù.</p>
</div>
<div class="section level2">
<h2 id="calculate-association-statistics">Calculate Association Statistics<a class="anchor" aria-label="anchor" href="#calculate-association-statistics"></a>
</h2>
<p>With the beam.data and beam.specs we can now calculate the
association statistics for all omics and all endpoints in the original
data and the included bootstrap resamples. This can take a long time
depending on the size of your data, and is a natural place to
parallelize by computing the statistics for the original data and each
bootstrap dataset separately and then recombining.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">beam.stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_beam_stats.html">compute_beam_stats</a></span><span class="op">(</span>beam.data<span class="op">=</span><span class="va">beam.data</span>, beam.specs<span class="op">=</span><span class="va">beam.specs</span><span class="op">)</span></span>
<span><span class="co">#&gt; Working on analysis 1 of 6 : Tue Jul  9 18:41:28 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 0 of 11 : Tue Jul  9 18:41:28 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 10 of 11 : Tue Jul  9 18:41:28 2024</span></span>
<span><span class="co">#&gt; Working on analysis 2 of 6 : Tue Jul  9 18:41:28 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 0 of 11 : Tue Jul  9 18:41:28 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 10 of 11 : Tue Jul  9 18:41:28 2024</span></span>
<span><span class="co">#&gt; Working on analysis 3 of 6 : Tue Jul  9 18:41:28 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 0 of 11 : Tue Jul  9 18:41:28 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 10 of 11 : Tue Jul  9 18:41:43 2024</span></span>
<span><span class="co">#&gt; Working on analysis 4 of 6 : Tue Jul  9 18:41:44 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 0 of 11 : Tue Jul  9 18:41:44 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 10 of 11 : Tue Jul  9 18:42:00 2024</span></span>
<span><span class="co">#&gt; Working on analysis 5 of 6 : Tue Jul  9 18:42:02 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 0 of 11 : Tue Jul  9 18:42:02 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 10 of 11 : Tue Jul  9 18:42:17 2024</span></span>
<span><span class="co">#&gt; Working on analysis 6 of 6 : Tue Jul  9 18:42:19 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 0 of 11 : Tue Jul  9 18:42:19 2024</span></span>
<span><span class="co">#&gt;   Working on bootstrap 10 of 11 : Tue Jul  9 18:42:37 2024</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">beam.stats</span></span>
<span><span class="co">#&gt; Contains 6 association estimate matrices with 10 bootstraps: </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of Lesion with MRD29 has dimensions 20 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of RNA with MRD29 has dimensions 20 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of Lesion with OS has dimensions 20 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of RNA with OS has dimensions 20 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of Lesion with EFS has dimensions 20 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of RNA with EFS has dimensions 20 x 11. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for Lesion with MRD29: </span></span>
<span><span class="co">#&gt;                               boot_0      boot_1      boot_2      boot_3</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss     -0.19093600 -0.18285072 -0.21417250 -0.20583809</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss     -0.19931154 -0.17555327 -0.20297360 -0.19104882</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss     -0.18613078 -0.17555327 -0.21375800 -0.18794477</span></span>
<span><span class="co">#&gt; ENSG00000148400_mutation -0.02983458 -0.03084684  0.05543433 -0.01137685</span></span>
<span><span class="co">#&gt; ENSG00000099810_loss     -0.13672414 -0.12616440 -0.16711568 -0.11536782</span></span>
<span><span class="co">#&gt;                                boot_4</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss     -0.156257583</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss     -0.210577002</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss     -0.163050513</span></span>
<span><span class="co">#&gt; ENSG00000148400_mutation -0.001250098</span></span>
<span><span class="co">#&gt; ENSG00000099810_loss     -0.090666306</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for RNA with MRD29: </span></span>
<span><span class="co">#&gt;                       boot_0       boot_1      boot_2       boot_3      boot_4</span></span>
<span><span class="co">#&gt; ENSG00000121410  0.045547007  0.004894953  0.11539090  0.091368164  0.12737124</span></span>
<span><span class="co">#&gt; ENSG00000148584 -0.043899068 -0.027183517 -0.03688248 -0.066976684 -0.06705541</span></span>
<span><span class="co">#&gt; ENSG00000175899  0.071592760  0.035710522  0.08890343  0.060304200  0.04032931</span></span>
<span><span class="co">#&gt; ENSG00000166535  0.000296431 -0.011542505 -0.07000542  0.003024411  0.03280950</span></span>
<span><span class="co">#&gt; ENSG00000184389  0.088934788  0.053931057  0.12269236  0.205216952  0.11770464</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for Lesion with OS: </span></span>
<span><span class="co">#&gt;                              boot_0     boot_1     boot_2       boot_3</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss      0.8932939 0.82087274  0.7774685  1.007661292</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss      0.9489993 0.84854596  0.8068623  1.045121725</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss      0.9563505 0.84854596  0.8400789  1.095728463</span></span>
<span><span class="co">#&gt; ENSG00000148400_mutation -0.0989406 0.04244735  0.1444102 -0.081025546</span></span>
<span><span class="co">#&gt; ENSG00000099810_loss      0.3328451 0.53394399 -0.1096147  0.003899453</span></span>
<span><span class="co">#&gt;                               boot_4</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss     1.116271049</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss     1.172808630</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss     1.139336521</span></span>
<span><span class="co">#&gt; ENSG00000148400_mutation 0.008572771</span></span>
<span><span class="co">#&gt; ENSG00000099810_loss     0.583670072</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for RNA with OS: </span></span>
<span><span class="co">#&gt;                      boot_0      boot_1      boot_2      boot_3       boot_4</span></span>
<span><span class="co">#&gt; ENSG00000121410 -0.06064136  0.08833406  0.19740235 -0.23032144  0.042520410</span></span>
<span><span class="co">#&gt; ENSG00000148584  0.05061313  0.19634699  0.13935280 -0.04349620 -0.308207853</span></span>
<span><span class="co">#&gt; ENSG00000175899  0.02594285  0.03512339  0.06672908 -0.11478157 -0.057850041</span></span>
<span><span class="co">#&gt; ENSG00000166535 -0.09119894 -0.16775731 -0.97860178 -0.05163522  0.036242601</span></span>
<span><span class="co">#&gt; ENSG00000184389 -0.13155891 -0.34544421 -0.27764175 -0.10397644  0.009883494</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for Lesion with EFS: </span></span>
<span><span class="co">#&gt;                                boot_0      boot_1       boot_2      boot_3</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss      0.311047946 0.192434740 -0.027644863  0.45783281</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss      0.355335350 0.212469747 -0.005443877  0.49058097</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss      0.358707580 0.212469747  0.010605097  0.52911969</span></span>
<span><span class="co">#&gt; ENSG00000148400_mutation -0.008535379 0.101993852 -0.185038910 -0.09593391</span></span>
<span><span class="co">#&gt; ENSG00000099810_loss     -0.015810223 0.006761143 -0.488404478 -0.18680785</span></span>
<span><span class="co">#&gt;                             boot_4</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss     0.6079674</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss     0.6584991</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss     0.6237131</span></span>
<span><span class="co">#&gt; ENSG00000148400_mutation 0.1068464</span></span>
<span><span class="co">#&gt; ENSG00000099810_loss     0.2182788</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for RNA with EFS: </span></span>
<span><span class="co">#&gt;                      boot_0     boot_1      boot_2     boot_3      boot_4</span></span>
<span><span class="co">#&gt; ENSG00000121410  0.05921417  0.1825319  0.34583467  0.1120400  0.07337122</span></span>
<span><span class="co">#&gt; ENSG00000148584  0.17012792  0.2792833  0.35866190  0.2208551 -0.50231550</span></span>
<span><span class="co">#&gt; ENSG00000175899 -0.11536335 -0.1158022 -0.02586119 -5.6659243 -3.05803752</span></span>
<span><span class="co">#&gt; ENSG00000166535  0.24454397  0.3871298  0.45451663  0.2155649  0.19137130</span></span>
<span><span class="co">#&gt; ENSG00000184389 -0.13690629 -0.2247154 -0.12818200 -0.2024793 -0.02224107</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Endpoint Data: </span></span>
<span><span class="co">#&gt;        MRD29   EFS    OS</span></span>
<span><span class="co">#&gt; PARASZ  0.00 3087+ 3087+</span></span>
<span><span class="co">#&gt; PARAYM  0.00 3399+ 3399+</span></span>
<span><span class="co">#&gt; PARCVM  0.00 2424+ 2424+</span></span>
<span><span class="co">#&gt; PAREGZ  0.49 3087+ 3087+</span></span>
<span><span class="co">#&gt; PARFDL  0.00 3075+ 3075+</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; BEAM Model Specifications: </span></span>
<span><span class="co">#&gt;           name    mtx                                         mdl</span></span>
<span><span class="co">#&gt; 1 Lesion.MRD29 Lesion    lm(MRD29~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 2    RNA.MRD29    RNA    lm(MRD29~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 3    Lesion.OS Lesion  coxphf2(OS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 4       RNA.OS    RNA  coxphf2(OS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 5   Lesion.EFS Lesion coxphf2(EFS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 6      RNA.EFS    RNA coxphf2(EFS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; BEAM data used to create Association Estimate Matrices: </span></span>
<span><span class="co">#&gt; main.data: 265 rows and 8 columns. </span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;            ID MRD29 RNA.clm Lesion.clm Lesion.id</span></span>
<span><span class="co">#&gt; PARASZ PARASZ  0.00      46         21    PARASZ</span></span>
<span><span class="co">#&gt; PARAYM PARAYM  0.00      47        108    PARAYM</span></span>
<span><span class="co">#&gt; PARCVM PARCVM  0.00      15         84    PARCVM</span></span>
<span><span class="co">#&gt; PAREGZ PAREGZ  0.49     170        142    PAREGZ</span></span>
<span><span class="co">#&gt; PARFDL PARFDL  0.00     184        175    PARFDL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; mtx.data: </span></span>
<span><span class="co">#&gt;   mtx.data Lesion: 265 columns linked to 265 rows of main.data. </span></span>
<span><span class="co">#&gt;   mtx.data RNA: 264 columns linked to 264 rows of main.data. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Lesion: </span></span>
<span><span class="co">#&gt;                          PARWNW PASXUC PATXNR PASYIS PATBGC</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000148400_mutation      1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000099810_loss          1      1      1      1      0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; RNA: </span></span>
<span><span class="co">#&gt;                    PARFIH     PARFPJ     PARFXJ     PARKLK     PARLJA</span></span>
<span><span class="co">#&gt; ENSG00000121410 0.4976379 0.29994795 1.13973142 0.32584690 0.35940872</span></span>
<span><span class="co">#&gt; ENSG00000148584 0.0000000 0.00000000 0.00000000 0.00000000 0.00000000</span></span>
<span><span class="co">#&gt; ENSG00000175899 0.0176369 0.11316706 0.02913480 0.00992041 0.02672925</span></span>
<span><span class="co">#&gt; ENSG00000166535 0.0000000 0.02261638 0.01278318 0.01735184 0.00000000</span></span>
<span><span class="co">#&gt; ENSG00000184389 0.1067503 1.46126661 0.32734246 0.56042150 0.49385225</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; mtx.anns: </span></span>
<span><span class="co">#&gt;   Lesion: 20 rows and 2 columns. </span></span>
<span><span class="co">#&gt;   RNA: 20 rows and 2 columns. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Lesion: </span></span>
<span><span class="co">#&gt;                         id            gene</span></span>
<span><span class="co">#&gt; 1     ENSG00000264545_loss ENSG00000264545</span></span>
<span><span class="co">#&gt; 2     ENSG00000147889_loss ENSG00000147889</span></span>
<span><span class="co">#&gt; 3     ENSG00000224854_loss ENSG00000224854</span></span>
<span><span class="co">#&gt; 4 ENSG00000148400_mutation ENSG00000148400</span></span>
<span><span class="co">#&gt; 5     ENSG00000099810_loss ENSG00000099810</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; RNA: </span></span>
<span><span class="co">#&gt;                id            gene</span></span>
<span><span class="co">#&gt; 1 ENSG00000121410 ENSG00000121410</span></span>
<span><span class="co">#&gt; 2 ENSG00000148584 ENSG00000148584</span></span>
<span><span class="co">#&gt; 3 ENSG00000175899 ENSG00000175899</span></span>
<span><span class="co">#&gt; 4 ENSG00000166535 ENSG00000166535</span></span>
<span><span class="co">#&gt; 5 ENSG00000184389 ENSG00000184389</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; anns.mtch: </span></span>
<span><span class="co">#&gt;   mtx.data mtx.anns id.clm nrow.mtx nrow.ann nrow.map</span></span>
<span><span class="co">#&gt; 1   Lesion   Lesion     id       20       20       20</span></span>
<span><span class="co">#&gt; 2      RNA      RNA     id       20       20       20</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; set.data: 40 rows assigning sets to data.mtx rows. </span></span>
<span><span class="co">#&gt;             set.id mtx.id               row.id</span></span>
<span><span class="co">#&gt; 9  ENSG00000081760    RNA      ENSG00000081760</span></span>
<span><span class="co">#&gt; 8  ENSG00000094914    RNA      ENSG00000094914</span></span>
<span><span class="co">#&gt; 25 ENSG00000099810 Lesion ENSG00000099810_loss</span></span>
<span><span class="co">#&gt; 19 ENSG00000099810    RNA      ENSG00000099810</span></span>
<span><span class="co">#&gt; 14 ENSG00000109576    RNA      ENSG00000109576</span></span>
<span><span class="co">#&gt; 10 ENSG00000114771    RNA      ENSG00000114771</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; set.anns:  rows of set annotations.</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; boot.index: 11 rows and 265 columns of bootstrap indices. </span></span>
<span><span class="co">#&gt;      [,1] [,2] [,3] [,4] [,5]</span></span>
<span><span class="co">#&gt; [1,]    1    2    3    4    5</span></span>
<span><span class="co">#&gt; [2,]  179   14  195  118  229</span></span>
<span><span class="co">#&gt; [3,]  108    8  114  261   29</span></span>
<span><span class="co">#&gt; [4,]   55   19  241  218  155</span></span>
<span><span class="co">#&gt; [5,]  145  200  211   69   46</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="calculate-set-and-feature-level-p-values">Calculate Set and Feature-Level P-Values<a class="anchor" aria-label="anchor" href="#calculate-set-and-feature-level-p-values"></a>
</h2>
<p>Now we can use the beam.stats to find the set and feature-level
p-values.</p>
<p>Set-level p-values: By default, the BEAMR set p-values are for the
integration of all available omics and endpoints. The result is a list,
and the BEAMR p-values are in the ‚Äúset.pvals‚Äù entry.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">set.pvals</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_set_pvalues.html">compute_set_pvalues</a></span><span class="op">(</span><span class="va">beam.stats</span><span class="op">)</span></span>
<span><span class="co">#&gt; Preparing bootstrap results for calculating feature set p-values: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt; Finding stats for each data matrix:Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   Finding stats for data matrix Lesion: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Finding features with with Lesion.MRD29 stats: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Finding features with with Lesion.OS stats: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Finding features with with Lesion.EFS stats: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   Finding stats for data matrix RNA: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Finding features with with RNA.MRD29 stats: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Finding features with with RNA.OS stats: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Finding features with with RNA.EFS stats: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt; Found 120 rows of stats: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt; Merging stats with feature-sets: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt; Merged feature-set stat rows: 120</span></span>
<span><span class="co">#&gt; Ordering and indexing feature sets: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   Cleaning up beam.stat matrices:Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Working on matrix 1 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Working on matrix 2 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Working on matrix 3 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Working on matrix 4 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Working on matrix 5 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;    Working on matrix 6 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   row.start row.end          set.id</span></span>
<span><span class="co">#&gt; 1         1       3 ENSG00000081760</span></span>
<span><span class="co">#&gt; 2         4       6 ENSG00000094914</span></span>
<span><span class="co">#&gt; 3         7      12 ENSG00000099810</span></span>
<span><span class="co">#&gt; 4        13      15 ENSG00000109576</span></span>
<span><span class="co">#&gt; 5        16      18 ENSG00000114771</span></span>
<span><span class="co">#&gt; 6        19      21 ENSG00000118017</span></span>
<span><span class="co">#&gt;   row.start row.end          set.id   stat.id</span></span>
<span><span class="co">#&gt; 1         1       1 ENSG00000081760   RNA.EFS</span></span>
<span><span class="co">#&gt; 2         2       2 ENSG00000081760 RNA.MRD29</span></span>
<span><span class="co">#&gt; 3         3       3 ENSG00000081760    RNA.OS</span></span>
<span><span class="co">#&gt; 4         4       4 ENSG00000094914   RNA.EFS</span></span>
<span><span class="co">#&gt; 5         5       5 ENSG00000094914 RNA.MRD29</span></span>
<span><span class="co">#&gt; 6         6       6 ENSG00000094914    RNA.OS</span></span>
<span><span class="co">#&gt;                mtx.row          set.id   stat.id          row.id</span></span>
<span><span class="co">#&gt; 62 RNA_ENSG00000081760 ENSG00000081760   RNA.EFS ENSG00000081760</span></span>
<span><span class="co">#&gt; 61 RNA_ENSG00000081760 ENSG00000081760 RNA.MRD29 ENSG00000081760</span></span>
<span><span class="co">#&gt; 63 RNA_ENSG00000081760 ENSG00000081760    RNA.OS ENSG00000081760</span></span>
<span><span class="co">#&gt; 66 RNA_ENSG00000094914 ENSG00000094914   RNA.EFS ENSG00000094914</span></span>
<span><span class="co">#&gt; 64 RNA_ENSG00000094914 ENSG00000094914 RNA.MRD29 ENSG00000094914</span></span>
<span><span class="co">#&gt; 65 RNA_ENSG00000094914 ENSG00000094914    RNA.OS ENSG00000094914</span></span>
<span><span class="co">#&gt; Computing p-value for feature set 1 of 34: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt; 13ENSG00000081760</span></span>
<span><span class="co">#&gt; Computing p-value for feature set 26 of 34: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt; 9496ENSG00000229835</span></span>
<span><span class="co">#&gt; Finished computing p-values at: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt; Minimum q-value is 0.106439437831269</span></span>
<span><span class="co">#&gt; Creating set p-value data frame.</span></span>
<span><span class="co">#&gt; Done creating data frame.</span></span>
<span><span class="co">#&gt; Creating list for output.</span></span></code></pre></div>
<p>Example of the results:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">set.pvals</span><span class="op">$</span><span class="va">set.pvals</span><span class="op">)</span></span>
<span><span class="co">#&gt;            set.id</span></span>
<span><span class="co">#&gt; 1 ENSG00000081760</span></span>
<span><span class="co">#&gt; 2 ENSG00000094914</span></span>
<span><span class="co">#&gt; 3 ENSG00000099810</span></span>
<span><span class="co">#&gt; 4 ENSG00000109576</span></span>
<span><span class="co">#&gt; 5 ENSG00000114771</span></span>
<span><span class="co">#&gt; 6 ENSG00000118017</span></span>
<span><span class="co">#&gt;                                                                           features</span></span>
<span><span class="co">#&gt; 1                                               1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 2                                               1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 3 1 Lesion.EFS | 1 Lesion.MRD29 | 1 Lesion.OS | 1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 4                                               1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 5                                               1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 6                                               1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt;   mean.from.center origin.to.center distance.ratio      p.set     q.set</span></span>
<span><span class="co">#&gt; 1              2.7        23.261819      8.6154884 0.02970528 0.1112937</span></span>
<span><span class="co">#&gt; 2              2.7         1.686417      0.6245987 0.40000000 0.2676141</span></span>
<span><span class="co">#&gt; 3              5.4         8.594254      1.5915285 0.30000000 0.2554498</span></span>
<span><span class="co">#&gt; 4              2.7         7.262608      2.6898548 0.10000000 0.1440999</span></span>
<span><span class="co">#&gt; 5              2.7        39.518499     14.6364811 0.01704578 0.1064394</span></span>
<span><span class="co">#&gt; 6              2.7         2.397169      0.8878403 0.40000000 0.2676141</span></span></code></pre></div>
<p>The <code><a href="../reference/compute_feature_pvalues.html">compute_feature_pvalues()</a></code> function returns a list
for each omics/endpoint pair with univariate associations for all omics
features.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">feat.pvals</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_feature_pvalues.html">compute_feature_pvalues</a></span><span class="op">(</span><span class="va">beam.stats</span><span class="op">)</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 1 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 20 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 2 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 20 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 3 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 20 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 4 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 20 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 5 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 20 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 6 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 20 features.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">feat.pvals</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Lesion.MRD29" "RNA.MRD29"    "Lesion.OS"    "RNA.OS"       "Lesion.EFS"  </span></span>
<span><span class="co">#&gt; [6] "RNA.EFS"</span></span></code></pre></div>
<p>Example of the results:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">feat.pvals</span><span class="op">$</span><span class="va">RNA.EFS</span><span class="op">)</span></span>
<span><span class="co">#&gt;                id            gene       beta         p         q</span></span>
<span><span class="co">#&gt; 1 ENSG00000081760 ENSG00000081760 0.39512059 0.1666667 0.1969697</span></span>
<span><span class="co">#&gt; 2 ENSG00000094914 ENSG00000094914 0.16586489 0.3333333 0.2888889</span></span>
<span><span class="co">#&gt; 3 ENSG00000099810 ENSG00000099810 0.42230413 0.1666667 0.1969697</span></span>
<span><span class="co">#&gt; 4 ENSG00000109576 ENSG00000109576 0.08833319 0.5000000 0.3823529</span></span>
<span><span class="co">#&gt; 5 ENSG00000114771 ENSG00000114771 0.14483645 0.1666667 0.1969697</span></span>
<span><span class="co">#&gt; 6 ENSG00000118017 ENSG00000118017 0.30517260 0.1666667 0.1969697</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="subset-the-results">Subset the Results<a class="anchor" aria-label="anchor" href="#subset-the-results"></a>
</h2>
<p>If you have large data, the beam.stats result file might be very
large, too large to open on a local machine. One option is to filter
this results file, which is needed to evaluate the results and plot. The
<code><a href="../reference/subset_beam_result.html">subset_beam_result()</a></code> file allows for filtering based on
many different criteria, such as specify a list fo features, a vector of
set.ids, a vector of endpoints, a vector of omics, and set or
feature-level p/q-value limits. There is an option ot use the
intersection or union of all specified criteria. Additionally, the
set-level p-values can be optionally recalculated, which should be done
if the number of omics or endpoints changes. The example below filters
the beam.stats object to the sets with the top ten smallest set-level
q-values.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">beam.stats.filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subset_beam_result.html">subset_beam_result</a></span><span class="op">(</span><span class="va">beam.stats</span>, <span class="va">set.pvals</span>, <span class="va">feat.pvals</span>, endpts<span class="op">=</span><span class="cn">NULL</span>,</span>
<span>                                      q.limit<span class="op">=</span><span class="fl">10</span>, intersect<span class="op">=</span><span class="cn">TRUE</span>, recalc<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Working on mtx.anns: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   Matching matrix 1 with annotations: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   Matching matrix 2 with annotations: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 1 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 6 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 2 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 7 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 3 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 6 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 4 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 7 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 5 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 6 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 6 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 7 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 1 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 6 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 2 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 7 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 3 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 6 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 4 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 7 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 5 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 6 features.</span></span>
<span><span class="co">#&gt; Computing feature p-values for stat matrix 6 of 6: Tue Jul  9 18:42:39 2024</span></span>
<span><span class="co">#&gt;   This matrix has 7 features.</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">beam.stats.filt</span></span>
<span><span class="co">#&gt; $beam.stats</span></span>
<span><span class="co">#&gt; Contains 6 association estimate matrices with 10 bootstraps: </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of Lesion with MRD29 has dimensions 6 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of RNA with MRD29 has dimensions 7 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of Lesion with OS has dimensions 6 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of RNA with OS has dimensions 7 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of Lesion with EFS has dimensions 6 x 11. </span></span>
<span><span class="co">#&gt;   Association Estimate Matrix of RNA with EFS has dimensions 7 x 11. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for Lesion with MRD29: </span></span>
<span><span class="co">#&gt;                           boot_0     boot_1     boot_2      boot_3      boot_4</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss -0.19093600 -0.1828507 -0.2141725 -0.20583809 -0.15625758</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss -0.19931154 -0.1755533 -0.2029736 -0.19104882 -0.21057700</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss -0.18613078 -0.1755533 -0.2137580 -0.18794477 -0.16305051</span></span>
<span><span class="co">#&gt; ENSG00000264801_loss -0.13821656 -0.1239665 -0.1639224 -0.11055410 -0.09439548</span></span>
<span><span class="co">#&gt; ENSG00000266446_loss -0.09222371 -0.0775523 -0.1134355 -0.06947515 -0.07033315</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for RNA with MRD29: </span></span>
<span><span class="co">#&gt;                      boot_0       boot_1       boot_2       boot_3      boot_4</span></span>
<span><span class="co">#&gt; ENSG00000128274  0.08720496  0.109390028  0.044897331  0.121861752  0.09987658</span></span>
<span><span class="co">#&gt; ENSG00000081760 -0.03607313  0.034956223 -0.089175927  0.081705062 -0.06983956</span></span>
<span><span class="co">#&gt; ENSG00000114771  0.07002589  0.099323229  0.001406326  0.070103099  0.09474937</span></span>
<span><span class="co">#&gt; ENSG00000109576 -0.12047921 -0.134490811 -0.099232706 -0.120472858 -0.12181346</span></span>
<span><span class="co">#&gt; ENSG00000224854  0.04198742 -0.002051417  0.007657997  0.002145128  0.04635030</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for Lesion with OS: </span></span>
<span><span class="co">#&gt;                         boot_0    boot_1     boot_2     boot_3    boot_4</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss 0.8932939 0.8208727  0.7774685 1.00766129 1.1162710</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss 0.9489993 0.8485460  0.8068623 1.04512173 1.1728086</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss 0.9563505 0.8485460  0.8400789 1.09572846 1.1393365</span></span>
<span><span class="co">#&gt; ENSG00000264801_loss 0.3681696 0.6134522 -0.1043398 0.04148775 0.6138597</span></span>
<span><span class="co">#&gt; ENSG00000266446_loss 0.5760150 0.8063569  0.1638686 0.92763980 0.6092065</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for RNA with OS: </span></span>
<span><span class="co">#&gt;                     boot_0      boot_1    boot_2     boot_3     boot_4</span></span>
<span><span class="co">#&gt; ENSG00000128274 0.16197577  0.09668843 0.1557670 0.23429709  0.2197898</span></span>
<span><span class="co">#&gt; ENSG00000081760 0.07025600  0.25418477 0.4471904 0.09829159  0.1214861</span></span>
<span><span class="co">#&gt; ENSG00000114771 0.15584991  0.14674332 0.2083145 0.16427747  0.1169206</span></span>
<span><span class="co">#&gt; ENSG00000109576 0.09189032 -0.57119111 0.2057833 0.13147781  0.2651223</span></span>
<span><span class="co">#&gt; ENSG00000224854 0.06835618  0.30907085 0.2299027 0.22839346 -0.1334088</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for Lesion with EFS: </span></span>
<span><span class="co">#&gt;                          boot_0     boot_1       boot_2     boot_3    boot_4</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss 0.31104795 0.19243474 -0.027644863  0.4578328 0.6079674</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss 0.35533535 0.21246975 -0.005443877  0.4905810 0.6584991</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss 0.35870758 0.21246975  0.010605097  0.5291197 0.6237131</span></span>
<span><span class="co">#&gt; ENSG00000264801_loss 0.01296298 0.06564000 -0.482688007 -0.1595180 0.2405010</span></span>
<span><span class="co">#&gt; ENSG00000266446_loss 0.13131513 0.09748747 -0.014785146  0.3507272 0.3510419</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Association Estimate Matrix for RNA with EFS: </span></span>
<span><span class="co">#&gt;                      boot_0      boot_1      boot_2     boot_3      boot_4</span></span>
<span><span class="co">#&gt; ENSG00000128274  0.20296851  0.44949340  0.22461005 0.05788957  0.07440748</span></span>
<span><span class="co">#&gt; ENSG00000081760  0.40551255  0.48707882  0.45943021 0.38063298  0.34116921</span></span>
<span><span class="co">#&gt; ENSG00000114771  0.13935591  0.11748291  0.13784936 0.12321800  0.13015927</span></span>
<span><span class="co">#&gt; ENSG00000109576  0.09502053 -0.16961948  0.01397912 0.07915553  0.19728078</span></span>
<span><span class="co">#&gt; ENSG00000224854 -0.03123313  0.07555118 -0.03490088 0.07155310 -0.17750700</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Example Endpoint Data: </span></span>
<span><span class="co">#&gt;        MRD29   EFS    OS</span></span>
<span><span class="co">#&gt; PARASZ  0.00 3087+ 3087+</span></span>
<span><span class="co">#&gt; PARAYM  0.00 3399+ 3399+</span></span>
<span><span class="co">#&gt; PARCVM  0.00 2424+ 2424+</span></span>
<span><span class="co">#&gt; PAREGZ  0.49 3087+ 3087+</span></span>
<span><span class="co">#&gt; PARFDL  0.00 3075+ 3075+</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; BEAM Model Specifications: </span></span>
<span><span class="co">#&gt;           name    mtx                                         mdl</span></span>
<span><span class="co">#&gt; 1 Lesion.MRD29 Lesion    lm(MRD29~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 2    RNA.MRD29    RNA    lm(MRD29~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 3    Lesion.OS Lesion  coxphf2(OS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 4       RNA.OS    RNA  coxphf2(OS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 5   Lesion.EFS Lesion coxphf2(EFS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 6      RNA.EFS    RNA coxphf2(EFS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; BEAM data used to create Association Estimate Matrices: </span></span>
<span><span class="co">#&gt; main.data: 265 rows and 8 columns. </span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt;            ID MRD29 RNA.clm Lesion.clm Lesion.id</span></span>
<span><span class="co">#&gt; PARASZ PARASZ  0.00      46         21    PARASZ</span></span>
<span><span class="co">#&gt; PARAYM PARAYM  0.00      47        108    PARAYM</span></span>
<span><span class="co">#&gt; PARCVM PARCVM  0.00      15         84    PARCVM</span></span>
<span><span class="co">#&gt; PAREGZ PAREGZ  0.49     170        142    PAREGZ</span></span>
<span><span class="co">#&gt; PARFDL PARFDL  0.00     184        175    PARFDL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; mtx.data: </span></span>
<span><span class="co">#&gt;   mtx.data Lesion: 265 columns linked to 265 rows of main.data. </span></span>
<span><span class="co">#&gt;   mtx.data RNA: 264 columns linked to 264 rows of main.data. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Lesion: </span></span>
<span><span class="co">#&gt;                      PARWNW PASXUC PATXNR PASYIS PATBGC</span></span>
<span><span class="co">#&gt; ENSG00000264545_loss      1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000147889_loss      1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000224854_loss      1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000264801_loss      1      1      1      1      0</span></span>
<span><span class="co">#&gt; ENSG00000266446_loss      1      1      1      0      0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; RNA: </span></span>
<span><span class="co">#&gt;                     PARFIH    PARFPJ    PARFXJ     PARKLK    PARLJA</span></span>
<span><span class="co">#&gt; ENSG00000128274 0.05672224 0.4579619 0.2620723 0.02147973 0.1392075</span></span>
<span><span class="co">#&gt; ENSG00000081760 1.35366204 1.6374711 1.4004286 1.72334012 1.2025118</span></span>
<span><span class="co">#&gt; ENSG00000114771 0.00000000 0.0000000 0.0000000 0.00000000 0.0000000</span></span>
<span><span class="co">#&gt; ENSG00000109576 2.15911246 0.2559237 2.1404842 0.92850298 1.3338817</span></span>
<span><span class="co">#&gt; ENSG00000224854 0.00000000 0.1323794 0.0000000 0.00000000 0.0000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; mtx.anns: </span></span>
<span><span class="co">#&gt;   Lesion: 6 rows and 2 columns. </span></span>
<span><span class="co">#&gt;   RNA: 7 rows and 2 columns. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Lesion: </span></span>
<span><span class="co">#&gt;                     id            gene</span></span>
<span><span class="co">#&gt; 1 ENSG00000264545_loss ENSG00000264545</span></span>
<span><span class="co">#&gt; 2 ENSG00000147889_loss ENSG00000147889</span></span>
<span><span class="co">#&gt; 3 ENSG00000224854_loss ENSG00000224854</span></span>
<span><span class="co">#&gt; 6 ENSG00000264801_loss ENSG00000264801</span></span>
<span><span class="co">#&gt; 9 ENSG00000266446_loss ENSG00000266446</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; RNA: </span></span>
<span><span class="co">#&gt;                 id            gene</span></span>
<span><span class="co">#&gt; 6  ENSG00000128274 ENSG00000128274</span></span>
<span><span class="co">#&gt; 9  ENSG00000081760 ENSG00000081760</span></span>
<span><span class="co">#&gt; 10 ENSG00000114771 ENSG00000114771</span></span>
<span><span class="co">#&gt; 14 ENSG00000109576 ENSG00000109576</span></span>
<span><span class="co">#&gt; 15 ENSG00000224854 ENSG00000224854</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; anns.mtch: </span></span>
<span><span class="co">#&gt;   mtx.data mtx.anns id.clm nrow.mtx nrow.ann nrow.map</span></span>
<span><span class="co">#&gt; 1   Lesion   Lesion     id        6        6        6</span></span>
<span><span class="co">#&gt; 2      RNA      RNA     id        7        7        7</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; set.data: 13 rows assigning sets to data.mtx rows. </span></span>
<span><span class="co">#&gt;             set.id mtx.id               row.id</span></span>
<span><span class="co">#&gt; 9  ENSG00000081760    RNA      ENSG00000081760</span></span>
<span><span class="co">#&gt; 14 ENSG00000109576    RNA      ENSG00000109576</span></span>
<span><span class="co">#&gt; 10 ENSG00000114771    RNA      ENSG00000114771</span></span>
<span><span class="co">#&gt; 6  ENSG00000128274    RNA      ENSG00000128274</span></span>
<span><span class="co">#&gt; 30 ENSG00000147883 Lesion ENSG00000147883_loss</span></span>
<span><span class="co">#&gt; 17 ENSG00000147883    RNA      ENSG00000147883</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; set.anns:  rows of set annotations.</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; boot.index: 11 rows and 265 columns of bootstrap indices. </span></span>
<span><span class="co">#&gt;      [,1] [,2] [,3] [,4] [,5]</span></span>
<span><span class="co">#&gt; [1,]    1    2    3    4    5</span></span>
<span><span class="co">#&gt; [2,]  179   14  195  118  229</span></span>
<span><span class="co">#&gt; [3,]  108    8  114  261   29</span></span>
<span><span class="co">#&gt; [4,]   55   19  241  218  155</span></span>
<span><span class="co">#&gt; [5,]  145  200  211   69   46</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $beam.set.pvals</span></span>
<span><span class="co">#&gt; $beam.set.pvals$set.pvals</span></span>
<span><span class="co">#&gt;             set.id</span></span>
<span><span class="co">#&gt; 1  ENSG00000081760</span></span>
<span><span class="co">#&gt; 4  ENSG00000109576</span></span>
<span><span class="co">#&gt; 5  ENSG00000114771</span></span>
<span><span class="co">#&gt; 8  ENSG00000128274</span></span>
<span><span class="co">#&gt; 9  ENSG00000147883</span></span>
<span><span class="co">#&gt; 10 ENSG00000147889</span></span>
<span><span class="co">#&gt; 23 ENSG00000224854</span></span>
<span><span class="co">#&gt; 30 ENSG00000264545</span></span>
<span><span class="co">#&gt; 31 ENSG00000264801</span></span>
<span><span class="co">#&gt; 33 ENSG00000266446</span></span>
<span><span class="co">#&gt;                                                                            features</span></span>
<span><span class="co">#&gt; 1                                                1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 4                                                1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 5                                                1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 8                                                1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 9  1 Lesion.EFS | 1 Lesion.MRD29 | 1 Lesion.OS | 1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 10 1 Lesion.EFS | 1 Lesion.MRD29 | 1 Lesion.OS | 1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 23 1 Lesion.EFS | 1 Lesion.MRD29 | 1 Lesion.OS | 1 RNA.EFS | 1 RNA.MRD29 | 1 RNA.OS</span></span>
<span><span class="co">#&gt; 30                                      1 Lesion.EFS | 1 Lesion.MRD29 | 1 Lesion.OS</span></span>
<span><span class="co">#&gt; 31                                      1 Lesion.EFS | 1 Lesion.MRD29 | 1 Lesion.OS</span></span>
<span><span class="co">#&gt; 33                                      1 Lesion.EFS | 1 Lesion.MRD29 | 1 Lesion.OS</span></span>
<span><span class="co">#&gt;    mean.from.center origin.to.center distance.ratio      p.set     q.set</span></span>
<span><span class="co">#&gt; 1               2.7        23.261819       8.615488 0.02970528 0.1112937</span></span>
<span><span class="co">#&gt; 4               2.7         7.262608       2.689855 0.10000000 0.1440999</span></span>
<span><span class="co">#&gt; 5               2.7        39.518499      14.636481 0.01704578 0.1064394</span></span>
<span><span class="co">#&gt; 8               2.7         6.541296       2.422702 0.10000000 0.1440999</span></span>
<span><span class="co">#&gt; 9               5.4        14.540832       2.692747 0.07445442 0.1440999</span></span>
<span><span class="co">#&gt; 10              5.4        71.196922      13.184615 0.01298854 0.1064394</span></span>
<span><span class="co">#&gt; 23              5.4        64.186229      11.886339 0.02357628 0.1104135</span></span>
<span><span class="co">#&gt; 30              2.7        60.083333      22.253086 0.01273073 0.1064394</span></span>
<span><span class="co">#&gt; 31              2.7         5.599363       2.073838 0.10000000 0.1440999</span></span>
<span><span class="co">#&gt; 33              2.7         5.851065       2.167061 0.09599200 0.1440999</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $beam.set.pvals$row.pvals</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $beam.set.pvals$set.mtch</span></span>
<span><span class="co">#&gt;                         mtx.row          set.id      stat.id</span></span>
<span><span class="co">#&gt; 62          RNA_ENSG00000081760 ENSG00000081760      RNA.EFS</span></span>
<span><span class="co">#&gt; 61          RNA_ENSG00000081760 ENSG00000081760    RNA.MRD29</span></span>
<span><span class="co">#&gt; 63          RNA_ENSG00000081760 ENSG00000081760       RNA.OS</span></span>
<span><span class="co">#&gt; 71          RNA_ENSG00000109576 ENSG00000109576      RNA.EFS</span></span>
<span><span class="co">#&gt; 70          RNA_ENSG00000109576 ENSG00000109576    RNA.MRD29</span></span>
<span><span class="co">#&gt; 72          RNA_ENSG00000109576 ENSG00000109576       RNA.OS</span></span>
<span><span class="co">#&gt; 74          RNA_ENSG00000114771 ENSG00000114771      RNA.EFS</span></span>
<span><span class="co">#&gt; 75          RNA_ENSG00000114771 ENSG00000114771    RNA.MRD29</span></span>
<span><span class="co">#&gt; 73          RNA_ENSG00000114771 ENSG00000114771       RNA.OS</span></span>
<span><span class="co">#&gt; 82          RNA_ENSG00000128274 ENSG00000128274      RNA.EFS</span></span>
<span><span class="co">#&gt; 84          RNA_ENSG00000128274 ENSG00000128274    RNA.MRD29</span></span>
<span><span class="co">#&gt; 83          RNA_ENSG00000128274 ENSG00000128274       RNA.OS</span></span>
<span><span class="co">#&gt; 4   Lesion_ENSG00000147883_loss ENSG00000147883   Lesion.EFS</span></span>
<span><span class="co">#&gt; 5   Lesion_ENSG00000147883_loss ENSG00000147883 Lesion.MRD29</span></span>
<span><span class="co">#&gt; 6   Lesion_ENSG00000147883_loss ENSG00000147883    Lesion.OS</span></span>
<span><span class="co">#&gt; 86          RNA_ENSG00000147883 ENSG00000147883      RNA.EFS</span></span>
<span><span class="co">#&gt; 87          RNA_ENSG00000147883 ENSG00000147883    RNA.MRD29</span></span>
<span><span class="co">#&gt; 85          RNA_ENSG00000147883 ENSG00000147883       RNA.OS</span></span>
<span><span class="co">#&gt; 8   Lesion_ENSG00000147889_loss ENSG00000147889   Lesion.EFS</span></span>
<span><span class="co">#&gt; 9   Lesion_ENSG00000147889_loss ENSG00000147889 Lesion.MRD29</span></span>
<span><span class="co">#&gt; 7   Lesion_ENSG00000147889_loss ENSG00000147889    Lesion.OS</span></span>
<span><span class="co">#&gt; 90          RNA_ENSG00000147889 ENSG00000147889      RNA.EFS</span></span>
<span><span class="co">#&gt; 88          RNA_ENSG00000147889 ENSG00000147889    RNA.MRD29</span></span>
<span><span class="co">#&gt; 89          RNA_ENSG00000147889 ENSG00000147889       RNA.OS</span></span>
<span><span class="co">#&gt; 26  Lesion_ENSG00000224854_loss ENSG00000224854   Lesion.EFS</span></span>
<span><span class="co">#&gt; 27  Lesion_ENSG00000224854_loss ENSG00000224854 Lesion.MRD29</span></span>
<span><span class="co">#&gt; 25  Lesion_ENSG00000224854_loss ENSG00000224854    Lesion.OS</span></span>
<span><span class="co">#&gt; 119         RNA_ENSG00000224854 ENSG00000224854      RNA.EFS</span></span>
<span><span class="co">#&gt; 118         RNA_ENSG00000224854 ENSG00000224854    RNA.MRD29</span></span>
<span><span class="co">#&gt; 120         RNA_ENSG00000224854 ENSG00000224854       RNA.OS</span></span>
<span><span class="co">#&gt; 47  Lesion_ENSG00000264545_loss ENSG00000264545   Lesion.EFS</span></span>
<span><span class="co">#&gt; 48  Lesion_ENSG00000264545_loss ENSG00000264545 Lesion.MRD29</span></span>
<span><span class="co">#&gt; 46  Lesion_ENSG00000264545_loss ENSG00000264545    Lesion.OS</span></span>
<span><span class="co">#&gt; 50  Lesion_ENSG00000264801_loss ENSG00000264801   Lesion.EFS</span></span>
<span><span class="co">#&gt; 51  Lesion_ENSG00000264801_loss ENSG00000264801 Lesion.MRD29</span></span>
<span><span class="co">#&gt; 49  Lesion_ENSG00000264801_loss ENSG00000264801    Lesion.OS</span></span>
<span><span class="co">#&gt; 57  Lesion_ENSG00000266446_loss ENSG00000266446   Lesion.EFS</span></span>
<span><span class="co">#&gt; 56  Lesion_ENSG00000266446_loss ENSG00000266446 Lesion.MRD29</span></span>
<span><span class="co">#&gt; 55  Lesion_ENSG00000266446_loss ENSG00000266446    Lesion.OS</span></span>
<span><span class="co">#&gt;                   row.id</span></span>
<span><span class="co">#&gt; 62       ENSG00000081760</span></span>
<span><span class="co">#&gt; 61       ENSG00000081760</span></span>
<span><span class="co">#&gt; 63       ENSG00000081760</span></span>
<span><span class="co">#&gt; 71       ENSG00000109576</span></span>
<span><span class="co">#&gt; 70       ENSG00000109576</span></span>
<span><span class="co">#&gt; 72       ENSG00000109576</span></span>
<span><span class="co">#&gt; 74       ENSG00000114771</span></span>
<span><span class="co">#&gt; 75       ENSG00000114771</span></span>
<span><span class="co">#&gt; 73       ENSG00000114771</span></span>
<span><span class="co">#&gt; 82       ENSG00000128274</span></span>
<span><span class="co">#&gt; 84       ENSG00000128274</span></span>
<span><span class="co">#&gt; 83       ENSG00000128274</span></span>
<span><span class="co">#&gt; 4   ENSG00000147883_loss</span></span>
<span><span class="co">#&gt; 5   ENSG00000147883_loss</span></span>
<span><span class="co">#&gt; 6   ENSG00000147883_loss</span></span>
<span><span class="co">#&gt; 86       ENSG00000147883</span></span>
<span><span class="co">#&gt; 87       ENSG00000147883</span></span>
<span><span class="co">#&gt; 85       ENSG00000147883</span></span>
<span><span class="co">#&gt; 8   ENSG00000147889_loss</span></span>
<span><span class="co">#&gt; 9   ENSG00000147889_loss</span></span>
<span><span class="co">#&gt; 7   ENSG00000147889_loss</span></span>
<span><span class="co">#&gt; 90       ENSG00000147889</span></span>
<span><span class="co">#&gt; 88       ENSG00000147889</span></span>
<span><span class="co">#&gt; 89       ENSG00000147889</span></span>
<span><span class="co">#&gt; 26  ENSG00000224854_loss</span></span>
<span><span class="co">#&gt; 27  ENSG00000224854_loss</span></span>
<span><span class="co">#&gt; 25  ENSG00000224854_loss</span></span>
<span><span class="co">#&gt; 119      ENSG00000224854</span></span>
<span><span class="co">#&gt; 118      ENSG00000224854</span></span>
<span><span class="co">#&gt; 120      ENSG00000224854</span></span>
<span><span class="co">#&gt; 47  ENSG00000264545_loss</span></span>
<span><span class="co">#&gt; 48  ENSG00000264545_loss</span></span>
<span><span class="co">#&gt; 46  ENSG00000264545_loss</span></span>
<span><span class="co">#&gt; 50  ENSG00000264801_loss</span></span>
<span><span class="co">#&gt; 51  ENSG00000264801_loss</span></span>
<span><span class="co">#&gt; 49  ENSG00000264801_loss</span></span>
<span><span class="co">#&gt; 57  ENSG00000266446_loss</span></span>
<span><span class="co">#&gt; 56  ENSG00000266446_loss</span></span>
<span><span class="co">#&gt; 55  ENSG00000266446_loss</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $beam.feat.pvals</span></span>
<span><span class="co">#&gt; $beam.feat.pvals$Lesion.MRD29</span></span>
<span><span class="co">#&gt;                     id            gene        beta         p         q</span></span>
<span><span class="co">#&gt; 1 ENSG00000147883_loss ENSG00000147883 -0.06887792 0.6666667 0.4074074</span></span>
<span><span class="co">#&gt; 2 ENSG00000147889_loss ENSG00000147889 -0.18938381 0.1666667 0.1527778</span></span>
<span><span class="co">#&gt; 3 ENSG00000224854_loss ENSG00000224854 -0.17833096 0.1666667 0.1527778</span></span>
<span><span class="co">#&gt; 4 ENSG00000264545_loss ENSG00000264545 -0.18609926 0.1666667 0.1527778</span></span>
<span><span class="co">#&gt; 5 ENSG00000264801_loss ENSG00000264801 -0.12821056 0.1666667 0.1527778</span></span>
<span><span class="co">#&gt; 6 ENSG00000266446_loss ENSG00000266446 -0.07479441 0.5000000 0.3666667</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $beam.feat.pvals$RNA.MRD29</span></span>
<span><span class="co">#&gt;                id            gene        beta         p         q</span></span>
<span><span class="co">#&gt; 1 ENSG00000081760 ENSG00000081760 -0.03973064 0.6666667 0.6666667</span></span>
<span><span class="co">#&gt; 2 ENSG00000109576 ENSG00000109576 -0.11641160 0.1666667 0.3333333</span></span>
<span><span class="co">#&gt; 3 ENSG00000114771 ENSG00000114771  0.08370487 0.1666667 0.3333333</span></span>
<span><span class="co">#&gt; 4 ENSG00000128274 ENSG00000128274  0.09529498 0.1666667 0.3333333</span></span>
<span><span class="co">#&gt; 5 ENSG00000147883 ENSG00000147883 -0.05407062 0.5000000 0.6000000</span></span>
<span><span class="co">#&gt; 6 ENSG00000147889 ENSG00000147889 -0.01545819 0.5000000 0.6000000</span></span>
<span><span class="co">#&gt; 7 ENSG00000224854 ENSG00000224854  0.02761206 0.8333333 0.7142857</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $beam.feat.pvals$Lesion.OS</span></span>
<span><span class="co">#&gt;                     id            gene      beta         p         q</span></span>
<span><span class="co">#&gt; 1 ENSG00000147883_loss ENSG00000147883 0.2881053 0.5000000 0.2777778</span></span>
<span><span class="co">#&gt; 2 ENSG00000147889_loss ENSG00000147889 0.9426508 0.1666667 0.1388889</span></span>
<span><span class="co">#&gt; 3 ENSG00000224854_loss ENSG00000224854 0.9488390 0.1666667 0.1388889</span></span>
<span><span class="co">#&gt; 4 ENSG00000264545_loss ENSG00000264545 0.8836309 0.1666667 0.1388889</span></span>
<span><span class="co">#&gt; 5 ENSG00000264801_loss ENSG00000264801 0.3449753 0.5000000 0.2777778</span></span>
<span><span class="co">#&gt; 6 ENSG00000266446_loss ENSG00000266446 0.5411791 0.1666667 0.1388889</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $beam.feat.pvals$RNA.OS</span></span>
<span><span class="co">#&gt;                id            gene       beta         p         q</span></span>
<span><span class="co">#&gt; 1 ENSG00000081760 ENSG00000081760 0.13510178 0.5000000 0.3888889</span></span>
<span><span class="co">#&gt; 2 ENSG00000109576 ENSG00000109576 0.11760327 0.6666667 0.4444444</span></span>
<span><span class="co">#&gt; 3 ENSG00000114771 ENSG00000114771 0.14751566 0.1666667 0.1944444</span></span>
<span><span class="co">#&gt; 4 ENSG00000128274 ENSG00000128274 0.23584130 0.1666667 0.1944444</span></span>
<span><span class="co">#&gt; 5 ENSG00000147883 ENSG00000147883 0.20069900 0.1666667 0.1944444</span></span>
<span><span class="co">#&gt; 6 ENSG00000147889 ENSG00000147889 0.20974254 0.1666667 0.1944444</span></span>
<span><span class="co">#&gt; 7 ENSG00000224854 ENSG00000224854 0.08683849 0.5000000 0.3888889</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $beam.feat.pvals$Lesion.EFS</span></span>
<span><span class="co">#&gt;                     id            gene        beta         p         q</span></span>
<span><span class="co">#&gt; 1 ENSG00000147883_loss ENSG00000147883  0.03354078 0.8333333 1.0000000</span></span>
<span><span class="co">#&gt; 2 ENSG00000147889_loss ENSG00000147889  0.34089214 0.3333333 0.6666667</span></span>
<span><span class="co">#&gt; 3 ENSG00000224854_loss ENSG00000224854  0.34309511 0.1666667 0.6666667</span></span>
<span><span class="co">#&gt; 4 ENSG00000264545_loss ENSG00000264545  0.29337356 0.3333333 0.6666667</span></span>
<span><span class="co">#&gt; 5 ENSG00000264801_loss ENSG00000264801 -0.03602487 1.0000000 1.0000000</span></span>
<span><span class="co">#&gt; 6 ENSG00000266446_loss ENSG00000266446  0.12867032 0.8333333 1.0000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $beam.feat.pvals$RNA.EFS</span></span>
<span><span class="co">#&gt;                id            gene        beta         p         q</span></span>
<span><span class="co">#&gt; 1 ENSG00000081760 ENSG00000081760  0.39512059 0.1666667 0.1944444</span></span>
<span><span class="co">#&gt; 2 ENSG00000109576 ENSG00000109576  0.08833319 0.5000000 0.3888889</span></span>
<span><span class="co">#&gt; 3 ENSG00000114771 ENSG00000114771  0.14483645 0.1666667 0.1944444</span></span>
<span><span class="co">#&gt; 4 ENSG00000128274 ENSG00000128274  0.22670353 0.1666667 0.1944444</span></span>
<span><span class="co">#&gt; 5 ENSG00000147883 ENSG00000147883  0.12287691 0.3333333 0.3111111</span></span>
<span><span class="co">#&gt; 6 ENSG00000147889 ENSG00000147889  0.27823411 0.1666667 0.1944444</span></span>
<span><span class="co">#&gt; 7 ENSG00000224854 ENSG00000224854 -0.02584083 0.8333333 0.5555556</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="plot-the-results">Plot the Results<a class="anchor" aria-label="anchor" href="#plot-the-results"></a>
</h2>
<p>Finally, we have some functions to visualize the results. First we
need to update ‚Äúbeam.specs‚Äù to include a ‚Äúplot‚Äù column. This can be done
using the <code><a href="../reference/prep_beam_plot.html">prep_beam_plot()</a></code> function.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot.specs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep_beam_plot.html">prep_beam_plot</a></span><span class="op">(</span>beam.data<span class="op">=</span><span class="va">beam.data</span>, beam.specs<span class="op">=</span><span class="va">beam.specs</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot.specs</span></span>
<span><span class="co">#&gt;           name    mtx                                         mdl</span></span>
<span><span class="co">#&gt; 1 Lesion.MRD29 Lesion    lm(MRD29~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 2    RNA.MRD29    RNA    lm(MRD29~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 3    Lesion.OS Lesion  coxphf2(OS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 4       RNA.OS    RNA  coxphf2(OS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 5   Lesion.EFS Lesion coxphf2(EFS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt; 6      RNA.EFS    RNA coxphf2(EFS~mtx.row,data=main.data,model=T)</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                         plot</span></span>
<span><span class="co">#&gt; 1                                                                                        ggplot2::ggplot(data=main.data)+ggplot2::geom_boxplot(aes(x=mtx.row.fac, y=MRD29)) + ggplot2::ylab('MRD29')+ggplot2::xlab('Lesion')</span></span>
<span><span class="co">#&gt; 2                                                                                                 ggplot2::ggplot(data=main.data)+ggplot2::geom_point(aes(x=mtx.row, y=MRD29)) + ggplot2::ylab('MRD29')+ggplot2::xlab('RNA')</span></span>
<span><span class="co">#&gt; 3   survminer::ggsurvplot(survfit(OS~mtx.row.fac, data=main.data), data=main.data, legend='right', legend.title='Lesion', legend.labs=c(paste0(levels(main.data$mtx.row.fac))), ylab='OS Probability', font.x=12, font.y=12)</span></span>
<span><span class="co">#&gt; 4       survminer::ggsurvplot(survfit(OS~(mtx.row&gt;median(mtx.row, na.rm=T)), data=main.data), data=main.data, legend='right', legend.title='RNA', legend.labs=c('High', 'Low'), ylab='OS Probability', font.x=12, font.y=12)</span></span>
<span><span class="co">#&gt; 5 survminer::ggsurvplot(survfit(EFS~mtx.row.fac, data=main.data), data=main.data, legend='right', legend.title='Lesion', legend.labs=c(paste0(levels(main.data$mtx.row.fac))), ylab='EFS Probability', font.x=12, font.y=12)</span></span>
<span><span class="co">#&gt; 6     survminer::ggsurvplot(survfit(EFS~(mtx.row&gt;median(mtx.row, na.rm=T)), data=main.data), data=main.data, legend='right', legend.title='RNA', legend.labs=c('High', 'Low'), ylab='EFS Probability', font.x=12, font.y=12)</span></span></code></pre></div>
<p>We can plot the top features (smallest feature-level p-value) for a
specified set using the <code><a href="../reference/plot_beam_clin.html">plot_beam_clin()</a></code> function.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_beam_clin.html">plot_beam_clin</a></span><span class="op">(</span>beam.result<span class="op">=</span><span class="va">beam.stats</span>, beam.specs<span class="op">=</span><span class="va">plot.specs</span>,</span>
<span>               beam.set.pvals <span class="op">=</span> <span class="va">set.pvals</span>, beam.feat.pvals<span class="op">=</span><span class="va">feat.pvals</span>,</span>
<span>               set.id<span class="op">=</span><span class="st">"ENSG00000099810"</span>, pair.type<span class="op">=</span><span class="cn">NULL</span>, number.pairs<span class="op">=</span><span class="fl">1</span>,</span>
<span>               n.col<span class="op">=</span><span class="fl">3</span>, n.row<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_point()`).</span></span></code></pre></div>
<p><img src="BEAMR_files/figure-html/unnamed-chunk-21-1.png" width="960"></p>
<p>We can also plot all of the clinical endpoint association plots for a
specified feature using the <code><a href="../reference/plot_feat_clin.html">plot_feat_clin()</a></code> function.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_feat_clin.html">plot_feat_clin</a></span><span class="op">(</span>beam.result<span class="op">=</span><span class="va">beam.stats</span>, beam.specs<span class="op">=</span><span class="va">plot.specs</span>,</span>
<span>               beam.set.pvals<span class="op">=</span><span class="va">set.pvals</span>, beam.feat.pvals<span class="op">=</span><span class="va">feat.pvals</span>,</span>
<span>               feat.id<span class="op">=</span><span class="st">"ENSG00000227443_loss"</span>, n.col<span class="op">=</span><span class="fl">2</span>, n.row<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="BEAMR_files/figure-html/unnamed-chunk-22-1.png" width="960"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Anna Eames Seffernick, Stanley Pounds, Xueyuan Cao.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
